docs/Development/Release-Process.md

```diff
@@ -0,0 +1,62 @@
+# Release Process
+
+From early on, Diamond was run as a master is stable release style. The belief was that master should always be stable and ready to deploy to production. This lead to the odd release system we currently have.
+
+## Version String
+
+The current version is generated by the bash script at https://github.com/python-diamond/Diamond/blob/master/version.sh
+
+The major and minor version numbers are generated based on the latest annotated tag.
+
+Cutting a new major/minor version is done via:
+
+```shell
+git tag -a v4.1 -m "foobar"
+git push origin --tags
+```
+
+From this point forward, the makefile will generate a version starting with 4.1
```

Comment:
I think we need some guidelines for versioning. How about x.x.x:
Major - Major features, possible backwards incompatible changes, etc.
Minor - New collectors/handlers
Patch - Minor bug fixes, collector tweaks

What do you think? Would the minor number bump up too often with this scheme?